From 4027891cf0b6f6bc16d55d37fa87e5d309fa8fcf Mon Sep 17 00:00:00 2001
From: Johannes Berg <johannes.berg@intel.com>
Date: Sun, 5 Feb 2012 19:05:45 +0200
Subject: [PATCH 18/75] P2P: Remove unneeded go_neg_peer check from PD

When the GO negotiation peer is assigned, the state also cannot be IDLE,
SEARCH, or LISTEN_ONLY.

Signed-off-by: Johannes Berg <johannes.berg@intel.com>
(cherry picked from commit e12b85d324e9692443d666728b5d078d652fabeb)
---
 src/p2p/p2p_pd.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/src/p2p/p2p_pd.c
+++ b/src/p2p/p2p_pd.c
@@ -362,9 +362,8 @@ int p2p_prov_disc_req(struct p2p_data *p
 	else
 		dev->flags &= ~P2P_DEV_PD_FOR_JOIN;
 
-	if (p2p->go_neg_peer ||
-	    (p2p->state != P2P_IDLE && p2p->state != P2P_SEARCH &&
-	     p2p->state != P2P_LISTEN_ONLY)) {
+	if (p2p->state != P2P_IDLE && p2p->state != P2P_SEARCH &&
+	    p2p->state != P2P_LISTEN_ONLY) {
 		wpa_msg(p2p->cfg->msg_ctx, MSG_DEBUG, "P2P: Busy with other "
 			"operations; postpone Provision Discovery Request "
 			"with " MACSTR " (config methods 0x%x)",
